src/KafkaFlow.IntegrationTests/Core/Middlewares/TriggerErrorSerializerMiddleware.cs

```diff
@@ -0,0 +1,24 @@
+﻿namespace KafkaFlow.IntegrationTests.Core.Middlewares
+{
+    using System;
+    using System.IO;
+    using System.Threading.Tasks;
+    using Confluent.Kafka;
+
+    public class TriggerErrorSerializerMiddleware : ISerializer
+    {
+        /// <inheritdoc/>
+        public Task SerializeAsync(object message, Stream output, ISerializerContext context)
+        {
+            var error = new Error(ErrorCode.BrokerNotAvailable);
+            throw new ProduceException<byte[], byte[]>(error, null);
+        }
+
+        /// <inheritdoc/>
+        public Task<object> DeserializeAsync(Stream input, Type type, ISerializerContext context)
+        {
+            var error = new Error(ErrorCode.BrokerNotAvailable);
+            throw new ProduceException<byte[], byte[]>(error, null);
+        }
+    }
+}
```

Comment:
why not use the Mock lib to do this?

Reply:
I'm not using Moq because this approach allows me to replicate accurately the codeflow of what would happen if an error is thrown in a middleware. I also believe we should stray away from using Moq, perhaps changing it for an alternative within the Framework, due to the recent scandals that happened related to it